src/movies.js

```diff
@@ -1,16 +1,58 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
-
+function getAllDirectors(arr) {
+  const pickDirector = arr.map((x) => x.director);
+  return pickDirector;
+}
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(arr) {
+  const sortedMovie = arr.filter((movie) => {
+    if (
+      movie.director === 'Steven Spielberg' &&
+      movie.genre.indexOf('Drama') > -1
+    ) {
+      return true;
+    }
+  });
```

Comment:
with `filter()` method, you do not have to actually return true. If the condition passes -> the element will be moved to new array. So for example, the below would be equivalent to what you have:

```
  const sortedMovie = arr.filter((movie) => {
    return (
      movie.director === 'Steven Spielberg' &&
      movie.genre.indexOf('Drama') > -1
    ) 
  });
```

To make it more shorter, if you code is only one line - when using arrow function you can skip curly brace and return 

```
  const sortedMovie = arr.filter((movie) => movie.director === 'Steven Spielberg' &&
      movie.genre.indexOf('Drama') > -1);
```

But please only use this shorter syntax if you understand it! ;)